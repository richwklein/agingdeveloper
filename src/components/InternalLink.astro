---
/**
 * @name InternalLink
 * Component used to render a link internal to the site.
 *
 * @param {Props} props - The properties of the link.
 * @return {Function} an Astro component factory function.
 *
 * @example
 * <InternalLinkProps to="tags">Tags</InternalLink>
 */

interface Props {
  to: string;
  title?: string;
  class?: string;
}

const { to, title, class: className } = Astro.props;
---

<a data-link="internal" target="_self" href={to} class:list={[className]} title={title}><slot /></a>
