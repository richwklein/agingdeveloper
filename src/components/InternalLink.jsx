import React from "react";
import MuiLink from "@mui/material/Link";
import {Link as GatsbyLink} from "gatsby";
import {LinkProps} from "../props";


/**
 * React component used to render a link internal to the site.
 *
 * @param {LinkProps} props - The to link and rest.
 * @return {React.ReactElement} - The react component
 *
 * @example
 * <InternalLinkProps to="tags">Tags</InternalLink>
 * @todo Remove prop spread
 */
const InternalLink = ({to, ...rest}) => (
  <MuiLink data-link="internal" component={GatsbyLink} to={to} {...rest} />
);

InternalLink.propTypes = LinkProps;

export default InternalLink;
