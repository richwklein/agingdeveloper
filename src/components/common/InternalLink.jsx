import React from "react";
import MuiLink from "@mui/material/Link";
import {Link as GatsbyLink} from "gatsby";

const InternalLink = React.forwardRef((props, ref) => (
  <MuiLink data-link="internal" component={GatsbyLink} ref={ref} {...props} />
));

InternalLink.displayName = "InternalLink";

export default InternalLink;
