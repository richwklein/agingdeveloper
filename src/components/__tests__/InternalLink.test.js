import React, {Fragment} from "react";
import {cleanup, render} from "@testing-library/react";
import InternalLink from "../InternalLink";

describe("InternalLink", () => {
  afterEach(cleanup);

  const renderComponent = () => {
    return render(
        <InternalLink
          to="https://example.com"
          data-testid="internal-link">
          <Fragment>Test Content</Fragment>
        </InternalLink>,
    );
  };

  test("that it renders correctly.", () => {
    const {getByTestId} = renderComponent();
    expect(getByTestId("internal-link")).toMatchSnapshot();
  });
});
