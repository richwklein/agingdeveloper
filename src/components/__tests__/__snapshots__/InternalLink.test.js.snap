// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InternalLink that it renders correctly. 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
  data-testid="internal-link"
  href="https://example.com"
>
  Test Content
</a>
`;
